auth/external.go

```diff
@@ -0,0 +1,63 @@
+package auth
+
+import (
+	"bytes"
+	"log"
+	"net/http"
+
+	"github.com/fabiolb/fabio/config"
+)
+
+type external struct {
+	endpoint          string
+	appendAuthHeaders []string
+	setAuthHeaders    []string
+}
+
+func newExternalAuth(cfg config.ExternalAuth) (AuthScheme, error) {
+	return &external{
+		endpoint:          cfg.Endpoint,
+		appendAuthHeaders: cfg.AppendAuthHeaders,
+		setAuthHeaders:    cfg.SetAuthHeaders,
+	}, nil
+}
+
+func (b *external) Authorized(request *http.Request, response http.ResponseWriter) AuthDecision {
+	client := &http.Client{CheckRedirect: func(req *http.Request, via []*http.Request) error { return http.ErrUseLastResponse }}
```

Comment:
This client should specify a timeout at least, and possibly a transport too depending on whether tls configurations would need to be overridden.